source/common/upstream/upstream_impl.h

```diff
@@ -1037,6 +1041,7 @@ class ClusterInfoImpl : public ClusterInfo,
   const uint32_t max_response_headers_count_;
   LoadBalancerType lb_type_;
   const envoy::config::cluster::v3::Cluster::DiscoveryType type_;
+  const envoy::config::cluster::v3::Cluster original_config_;
```

Comment:
I'm not sure why this was added here but I think this should be specific to DFP and not leak into the common code.

Reply:
it's used for sub cluster inherit cluster config from the parent DFP cluster.
I'll try another proper way, thank you.

Reply:
Changed to add the `cluster` methods:
https://github.com/envoyproxy/envoy/pull/26420/files#diff-88c4ee0e7897570d35304eca6746d03781c2eacf20a84557b2c1e8df1c115e18R1201-R1210

1. we save the `DFP cluster` into `cluster_info` when initialize the `cluster` in dynamic_forward_proxy/cluster.cc:
https://github.com/envoyproxy/envoy/pull/26420/files#diff-8f0201f1ad64a4d5abf5d3cfe3388d19feea791341f62ebc32064622267da65aR434
2. then, we can get the DFP cluster in worker thread.